strawberryfields/plot.py

```diff
@@ -0,0 +1,152 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+This module provides tools to visualize the state in various interactive
+ways using Plot.ly
+"""
+from copy import deepcopy
+
+import numpy as np
+
+plotly_error = (
+    "Plot.ly required for using this function. It can be installed using pip install "
+    "plotly or visiting https://plot.ly/python/getting-started/#installation"
+)
+
+try:
+    import plotly.io as pio
+except ImportError as e:
+    raise(plotly_error) from e
```

Comment:
It would be good to determine when this exception is raised. For example, we don't want this to be raised when SF is imported; we only want it to be raised if/when the user tries calling the `plot_wigner` function. Perhaps we could define a function:

```python
def _get_plotly():
    try:
        import plotly.io as pio
    except ImportError as e:
        raise(plotly_error) from e
    return pio
```

Then, within the plotting functions, we could do:

```python
def plot_wigner(state, p_axis, q_axis, renderer="browser", wire=0, dq=0.1):
    """Plot the Wigner function with Plot.ly"""
    pio = _get_plotly()
```

Reply:
I like this solution. Thanks @josh146!